src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -231,6 +275,45 @@ internal RegionVersionOne CreateRegion(Region v2Region)
             return region;
         }
 
+        internal ReplacementVersionOne CreateReplacement(Replacement v2Replacement, Uri uri)
+        {
+            ReplacementVersionOne replacement = null;
+
+            if (v2Replacement != null)
+            {
+                replacement = new ReplacementVersionOne();
+                FileContent insertedContent = v2Replacement.InsertedContent;
+
+                if (insertedContent != null)
+                {
+                    if (insertedContent.Text != null)
```

Comment:
Bug: As the code stands, you will try (and possibly fail) to use the inserted text, even if the binary version is available to save you.

The logic should be:
```
IF insertedContent != null THEN
    IF the binary version exists
        Use it
    ELSE
        IF encoding is available
            Convert the text and use the result.
        ELSE
            throw
        END
    END
END
```

Reply:
Fixed